File: src/RadioMdl.py
"""
RadioMdl - A Python module for radio modeling calculations
"""

# Constants
k_boltz = 1.380649e-23  # Boltzmann's constant in J/K
rad = 3.141592653589793 / 180  # degree to radian conversion factor
speed_c = 3e8  # speed of light in m/s

# Note: We don't import other modules here to avoid circular imports
# Instead, import them directly in your scripts when needed

# Export constants and basic functions
__all__ = [
    'k_boltz',
    'rad',
    'speed_c',
    # Classes and functions should be imported directly from their modules
    # 'Antenna', 'Instrument', 'Observation', 'Constellation', 'Trajectory',
    # 'estim_temp', 'estim_casA_flux', 'power_to_temperature',
    # 'antenna_mdl_ITU', 'sat_link_budget', 'model_observed_temp'
]

File: Julia/src/RadioMdl.jl
module RadioMdl

export Antenna,
       antenna_mdl_ITU,
       Constellation,
       estim_casA_flux,
       estim_temp,
       estim_virgoA_flux,
       free_space_loss,
       get_center_freq_chans,
       get_def_angles,
       get_result,
       get_sats_name,
       get_sats_names_at_time,
       get_sat_traj,
       get_slice_gain,
       get_time_stamps,
       get_traj,
       Instrument,
       map_sphere,
       model_observed_temp!,
       Observation,
       power_to_temperature,
       sat_link_budget,
       simple_link_budget,
       Trajectory,
       temperature_to_flux,
       temperature_to_power

using Arrow
using DataFrames
using Dates
using DelimitedFiles
using Interpolations
using Trapz
using .Threads

""" Boltzman's constant in J/K"""
const k_boltz = 1.380649e-23

""" degree to radian conversion factor """
const rad = π/180

""" speed of light in m/s """
const speed_c = 3e8

include("astro_mdl.jl")
include("types.jl")
include("io.jl")
include("antenna_pattern.jl")
include("coord_frames.jl")
include("sat_mdl.jl")
include("obs_mdl.jl")

end # module RadioMdl
